examples/embeddable_examples/public/book/book_embeddable.tsx

```diff
@@ -0,0 +1,123 @@
+ * under the License.
+ */
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Subscription } from 'rxjs';
+import {
+  Embeddable,
+  EmbeddableInput,
+  IContainer,
+  EmbeddableOutput,
+  SavedObjectEmbeddableInput,
+  AttributeService,
+} from '../../../../src/plugins/embeddable/public';
+import { BookSavedObjectAttributes } from '../../common';
+import { BookEmbeddableComponent } from './book_component';
+
+export const BOOK_EMBEDDABLE = 'book';
+export type BookEmbeddableInput = BookByValueInput | BookByReferenceInput;
+export interface BookEmbeddableOutput extends EmbeddableOutput {
+  hasMatch: boolean;
+  attributes: BookSavedObjectAttributes;
+}
+
+interface BookInheritedInput extends EmbeddableInput {
+  search?: string;
+}
+
+export type BookByValueInput = { attributes: BookSavedObjectAttributes } & BookInheritedInput;
+export type BookByReferenceInput = SavedObjectEmbeddableInput & BookInheritedInput;
+
+/**
+ * Returns whether any attributes contain the search string.  If search is empty, true is returned. If
+ * there are no savedAttributes, false is returned.
+ * @param search - the search string
+ * @param savedAttributes - the saved object attributes for the saved object with id `input.savedObjectId`
+ */
+function getHasMatch(search?: string, savedAttributes?: BookSavedObjectAttributes): boolean {
+  if (!search) return true;
+  if (!savedAttributes) return false;
+  return Boolean(
+    (savedAttributes.author && savedAttributes.author.match(search)) ||
+      (savedAttributes.title && savedAttributes.title.match(search))
+  );
+}
+
+export class BookEmbeddable extends Embeddable<BookEmbeddableInput, BookEmbeddableOutput> {
+  public readonly type = BOOK_EMBEDDABLE;
+  private subscription: Subscription;
+  private node?: HTMLElement;
+  private savedObjectId?: string;
+  private attributes?: BookSavedObjectAttributes;
+
+  constructor(
+    initialInput: BookEmbeddableInput,
+    private attributeService: AttributeService<
+      BookSavedObjectAttributes,
+      BookByValueInput,
+      BookByReferenceInput
+    >,
+    {
+      parent,
+    }: {
+      parent?: IContainer;
+    }
+  ) {
+    super(initialInput, {} as BookEmbeddableOutput, parent);
+
+    this.subscription = this.getInput$().subscribe(async () => {
+      const savedObjectId = (this.getInput() as BookByReferenceInput).savedObjectId;
+      const attributes = (this.getInput() as BookByValueInput).attributes;
+      if (this.attributes !== attributes || this.savedObjectId !== savedObjectId) {
+        this.savedObjectId = savedObjectId;
+        this.reload();
+      } else {
+        this.updateOutput({
+          attributes: this.attributes,
+          hasMatch: getHasMatch(this.input.search, this.attributes),
+        });
+      }
+    });
+  }
+
+  public render(node: HTMLElement) {
+    this.node = node;
+    if (this.node) {
+      ReactDOM.unmountComponentAtNode(this.node);
```

Comment:
Should this line come before assigning a new node?

Reply:
Will look into this more... It's inconsistent inside the examples. `list_container` has it this way around, but `searchable_list_container` has it come after. It looks to me like it accomplishes the same thing, because this.node is only used on destroy.

Reply:
I think it was just a typo. I fixed it in `list_container`: https://github.com/elastic/kibana/pull/67783/files#r435423657